src/librustc_expand/mbe/macro_rules.rs

```diff
@@ -126,10 +143,11 @@ fn emit_frag_parse_err(
 impl<'a> ParserAnyMacro<'a> {
     crate fn make(mut self: Box<ParserAnyMacro<'a>>, kind: AstFragmentKind) -> AstFragment {
         let ParserAnyMacro { site_span, macro_ident, ref mut parser, arm_span } = *self;
+        let snapshot = &mut parser.clone();
```

Comment:
This is the main trade-off in this logic, but it doesn't seem all too costly.

Reply:
Can we perf it? I'm intrigued to see the effects of this change in the `syn` case.

Reply:
Perfbot says it was a win; I'm inclined not to trust that, but I don't think it's a regression either based on that data.